### PR TITLE
Automatically generate i18n javascript files for react-intl when the serve starts up.

### PR DESCRIPTION
**Pull request by JasonYCHuang**
This PR is for https://github.com/shakacode/react-webpack-rails-tutorial/pull/340
We need i18n_dir & default_locale in ReactOnRails configuration.

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/642)

### CODE REVIEW

**Comment by justin808** [comment]
we want this pr to include i18n.rake

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-:-KYfUx4z7tOs3KB0U6pN:b-7kqzp)*

**Comment by justin808** [comment]
@JasonYCHuang How's this PR going?

**Comment by JasonYCHuang** [comment]
@justin808 I am occupied recently, and will finish this during Xmas or New year holidays.

**Comment by justin808** [comment]
@jasonychuang Looks potentially super amazing! We need updates to the docs and a CHANGELOG.md, as well as tests that demonstrate and verify locals_to_js.rb.

This is really a huge addition to the project! :clap: :tada: 

---

Reviewed 2 of 5 files at r2.
Review status: 2 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-:-K_1oBvNWMxwTaYd1Leh:bff4pl6)*

**Comment by JasonYCHuang** [comment]
No problem, will update it later.

---

Review status: 2 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-:-K_3ditdEELKi2Pc123O:blj4p2t)*

**Comment by JasonYCHuang** [comment]
Update Readme, changelog, and add test

---

Review status: 2 of 8 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-:-K_K5jBd1no7J0AI5z3H:b-cdqryq)*

**Comment by justin808** [comment]
Looking super amazing. I left some comments.

---

Reviewed 2 of 5 files at r2, 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[README.md, line 173 at r3](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_RvQydqFBV_TE6tNqT:-K_RvQydqFBV_TE6tNqU:b99umf2) ([raw file](https://github.com/shakacode/react_on_rails/blob/2870647c5bc212e1ccc17afe7fd43fc7c77fd0bd/README.md#L173)):*
> ```Markdown
>   ```
> 
> ### I18n
> ```

Let's have a brief overview of the topic of i18n and have a pointer to a detailed doc in `/docs/basics`.

`/spec/dummy` should include a very simple demonstration that this technique works, as well as a test.

Possibly /docs/basics could refer to that.

---

*[lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt, line 70 at r3](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_Rv9War1LMVzgtepV_:-K_Rv9War1LMVzgtepVa:b-b6q56o) ([raw file](https://github.com/shakacode/react_on_rails/blob/2870647c5bc212e1ccc17afe7fd43fc7c77fd0bd/lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt#L70)):*
> ```
>   config.default_locale = "en"
>   # The location of rails locales
>   config.rails_locales_path = Rails.application.config.i18n.load_path
> ```

Why do we need this if Rails has this param? Could we always just read from the Rails param?

---

*[lib/react_on_rails/configuration.rb, line 110 at r3](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_RvGxdWlBZcTZfzfLk:-K_RvGxdWlBZcTZfzfLl:b-b6q56o) ([raw file](https://github.com/shakacode/react_on_rails/blob/2870647c5bc212e1ccc17afe7fd43fc7c77fd0bd/lib/react_on_rails/configuration.rb#L110)):*
> ```Ruby
>       self.i18n_dir = i18n_dir
>       self.default_locale = default_locale
>       self.rails_locales_path = rails_locales_path
> ```

Why do we need this if Rails has this param? Could we always just read from the Rails param?

---

*[spec/react_on_rails/locales_to_js_spec.rb, line 19 at r3](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_RvLTWh8Cnfdoqlda7:-K_RvLTWh8Cnfdoqlda8:b-ox54hp) ([raw file](https://github.com/shakacode/react_on_rails/blob/2870647c5bc212e1ccc17afe7fd43fc7c77fd0bd/spec/react_on_rails/locales_to_js_spec.rb#L19)):*
> ```Ruby
> 
>       files = Dir["#{i18n_dir.to_path}/*"].map { |p| Pathname.new(p).basename.to_s }
>       expect(files).to include("translations.js", "default.js")
> ```

We need to take a look inside the file to see that something confirms that just the files were not "touched".

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-:-K_RvpaBFK09XHt6JuZ8:b6l49vs)*

**Comment by justin808** [comment]
@JasonYCHuang I look forward to getting this one merged!

**Comment by JasonYCHuang** [comment]
I didn't realize that you already left the comment till your last comment.
I will work on it this weekend

**Comment by JasonYCHuang** [comment]
@justin808 
I have a problem on working in `spec/dummy`.
When I click server-rendering links, there will be error message: `Error prerendering in react_on_rails. See server logs.`
image

In client-rendering pages, there is only html content, but no react_on_rails content.
image

This problem is still exist if I switch to the latest master commit: `f929478`.
My node is `v7.0.0`.

```
➜  react_on_rails git:(master) $ cd spec/dummy
➜  dummy git:(master) $ bundle && npm install
...
➜  dummy git:(master) $ foreman start
```

Then, go to `http://localhost:3000/`, and I got error:
```
11:36:25 rails.1 | Started GET "/" for 127.0.0.1 at 2017-01-07 11:36:25 +0100
11:36:25 rails.1 | Processing by PagesController#index as HTML
11:36:25 rails.1 |   Rendered pages/_header.erb (3.8ms)
11:36:26 rails.1 | ERROR when compiling base_js_code! See file tmp/base_js_code.js to correlate line numbers of error. Error is
11:36:26 rails.1 |
11:36:26 rails.1 | Error: Cannot find module "react-on-rails"
11:36:26 rails.1 |
11:36:26 rails.1 | webpackMissingModule ((execjs):8232:86)
11:36:26 rails.1 | Object.(execjs) (<anonymous>:8232:177)
11:36:26 rails.1 | __webpack_require__ ((execjs):55:30)
11:36:26 rails.1 | Object.(execjs) (<anonymous>:83:19)
11:36:26 rails.1 | __webpack_require__ ((execjs):55:30)
11:36:26 rails.1 | (execjs):75:18
11:36:26 rails.1 | (execjs):78:10
...
```

Did I miss anything?

**Comment by justin808** [comment]
Please rebase on top of master.

**Comment by justin808** [comment]
I'd like to create a release today.

**Comment by JasonYCHuang** [comment]
Review status: 1 of 12 files reviewed at latest revision, 4 unresolved discussions.

---

*[README.md, line 173 at r3](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_RvQydqFBV_TE6tNqT:-K_uhjR9OrkyLtNTAVjG:bgiz8e) ([raw file](https://github.com/shakacode/react_on_rails/blob/2870647c5bc212e1ccc17afe7fd43fc7c77fd0bd/README.md#L173)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Let's have a brief overview of the topic of i18n and have a pointer to a detailed doc in `/docs/basics`.

`/spec/dummy` should include a very simple demonstration that this technique works, as well as a test.

Possibly /docs/basics could refer to that.


</blockquote></details>

Move i18n summary to `/docs/basics/i18n.md`.

As to `/spec/dummy`, I still can not find away to solve this.

---

*[lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt, line 70 at r3](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_Rv9War1LMVzgtepV_:-K_uiG5laBJeb6E44pZZ:b-kpq3hh) ([raw file](https://github.com/shakacode/react_on_rails/blob/2870647c5bc212e1ccc17afe7fd43fc7c77fd0bd/lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt#L70)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Why do we need this if Rails has this param? Could we always just read from the Rails param?
</blockquote></details>

remove `config.default_locale` & `config.rails_locales_path`, but keep `config.i18n_dir` since we need this to assign the destination for Javascript i18n.

---

*[lib/react_on_rails/configuration.rb, line 110 at r3](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_RvGxdWlBZcTZfzfLk:-K_uiinm65ksqkFGs52o:b-6jxh55) ([raw file](https://github.com/shakacode/react_on_rails/blob/2870647c5bc212e1ccc17afe7fd43fc7c77fd0bd/lib/react_on_rails/configuration.rb#L110)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Why do we need this if Rails has this param? Could we always just read from the Rails param?
</blockquote></details>

Keep `config.i18n_dir` since we need this to assign the destination for Javascript i18n.

---

*[spec/react_on_rails/locales_to_js_spec.rb, line 19 at r3](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_RvLTWh8Cnfdoqlda7:-K_uio2wnZkCctfvJori:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/2870647c5bc212e1ccc17afe7fd43fc7c77fd0bd/spec/react_on_rails/locales_to_js_spec.rb#L19)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

We need to take a look inside the file to see that something confirms that just the files were not "touched".
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642)*

**Comment by justin808** [comment]
Just a few small comments to address.

---

Reviewed 11 of 11 files at r4.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[.byebug_history, line 1 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vm_3XGHoLImEBdXfY:-K_vm_3XGHoLImEBdXfZ:b-3zcs6l) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/.byebug_history#L1)):*
> ```
> exit
> ```

@jasonychuang Please remove this file.

---

*[docs/basics/i18n.md, line 7 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vn1K4OVfl1OEQBZNi:-K_vn1K4OVfl1OEQBZNj:belfytt) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/docs/basics/i18n.md#L7)):*
> ```Markdown
> You can refer to [react-webpack-rails-tutorial](https://github.com/shakacode/react-webpack-rails-tutorial) for a complete example.
> 
> 1. Add `react-intl` to `client/package.json`, and remember to `bundle && npm install`
> ```

and `intl`

??

---

*[docs/basics/i18n.md, line 32 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vn800lQ9xOizAArbW:-K_vn800lQ9xOizAArbX:b-oc48r7) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/docs/basics/i18n.md#L32)):*
> ```Markdown
>   ```
> 
> 3. ReactOnRails uses locale files as you did before in Rails: `config/locales/*.yml`. Therefore, you don't need to create additional local files.
> ```

Could you only use the files that react-intl wants? and not use the Rails files?

---

*[docs/basics/i18n.md, line 37 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vnjQvDeJcmvpcWuri:-K_vnjQvDeJcmvpcWurj:bxlt50k) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/docs/basics/i18n.md#L37)):*
> ```Markdown
> 
>   ```ruby
>   # Replace the following line to the location where you keep translation.js & default.js.
> ```

Where do these 2 files come from?

---

*[docs/basics/i18n.md, line 41 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vnyYRqBAFDfDJ7_aj:-K_vnyYRqBAFDfDJ7_ak:bw6s54m) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/docs/basics/i18n.md#L41)):*
> ```Markdown
>   ```
> 
> 5. Add following lines to `config/application.rb`, this will help you to generate `translations.js` & `default.js` automatically when you starts the server.
> ```

We could do a quick check to see if the generated file is older than any of the locale files.

---

*[lib/react_on_rails/locales_to_js.rb, line 71 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vpVq_hYLOV4zFw1z_:-K_vpVq_hYLOV4zFw1za:b-6zq8nb) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/lib/react_on_rails/locales_to_js.rb#L71)):*
> ```Ruby
> 
>     def set_locale_files
>       Rails.application.config.i18n.load_path
> ```

do we need "set" in the method name?

---

*[lib/react_on_rails/locales_to_js.rb, line 74 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vpRhBp-3kacFdcdya:-K_vpRhBp-3kacFdcdyb:b-6zq8nb) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/lib/react_on_rails/locales_to_js.rb#L74)):*
> ```Ruby
>     end
> 
>     def set_default_locale
> ```

do we need "set" in the method name?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-:-K_vpl-r_lkFw5bwMDJR:b-ju13xa)*

**Comment by JasonYCHuang** [comment]
Review status: 7 of 11 files reviewed at latest revision, 9 unresolved discussions.

---

*[docs/basics/i18n.md, line 7 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vn1K4OVfl1OEQBZNi:-K_y9-zqLJHp6YeJoWwR:b-29fq7w) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/docs/basics/i18n.md#L7)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

and `intl`

??
</blockquote></details>

update description.

---

*[docs/basics/i18n.md, line 32 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vn800lQ9xOizAArbW:-K_y96BTZyUGFYRp3tRh:bctc0ya) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/docs/basics/i18n.md#L32)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Could you only use the files that react-intl wants? and not use the Rails files?
</blockquote></details>

Not sure I understand your questions.

`react-intl` wants locale files in json format, and I generate `translations.js` & `default.js`. Their sources are rails locale files.

I update the summary, and hope this will be more clear.

---

*[docs/basics/i18n.md, line 37 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vnjQvDeJcmvpcWuri:-K_yA4gfX7pcnfiOYZ6T:bb36625) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/docs/basics/i18n.md#L37)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Where do these 2 files come from?
</blockquote></details>

 These 2 files are generated by `lib/react_on_rails/locales_to_js.rb` in this PR.

---

*[docs/basics/i18n.md, line 41 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vnyYRqBAFDfDJ7_aj:-K_yALkZxN1hEq6dJe7u:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/docs/basics/i18n.md#L41)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

We could do a quick check to see if the generated file is older than any of the locale files.
</blockquote></details>

Done.

---

*[lib/react_on_rails/locales_to_js.rb, line 71 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vpVq_hYLOV4zFw1z_:-K_yAOlLENvlE0phcb18:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/lib/react_on_rails/locales_to_js.rb#L71)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

do we need "set" in the method name?
</blockquote></details>

Done.

---

*[lib/react_on_rails/locales_to_js.rb, line 74 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vpRhBp-3kacFdcdya:-K_yAPCYVG-FNBAiszfc:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/lib/react_on_rails/locales_to_js.rb#L74)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

do we need "set" in the method name?
</blockquote></details>

Done.

---

*[.byebug_history, line 1 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-K_vm_3XGHoLImEBdXfY:-K_yAQQwUKeTCEfAt_c9:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/.byebug_history#L1)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

@jasonychuang Please remove this file.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642)*

**Comment by justin808** [comment]
Please either merge or rebase to address the merge conflict.

**Comment by justin808** [comment]
Reviewed 5 of 5 files at r5.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[README.md, line 175 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-Ka1Iv86lD15HMnr-y6W:-Ka1Iv86lD15HMnr-y6X:b-6kd1gt) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/README.md#L175)):*
> ```Markdown
> ### I18n
> 
> You can enable the i18n functionality with [react-intl](https://github.com/yahoo/react-intl). ReactOnRails also converts traditional Rails locale files, `*.yml`, to required javascript files, `translations.js` & `default.js`, automatically.
> ```

I think we should consider mentioning that we support an optional automatic conversion of Rails *.yml files.

---

*[docs/basics/i18n.md, line 32 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-Ka1JJFTgRRCnVWxx3AL:-Ka1JJFTgRRCnVWxx3AM:bbkt9mw) ([raw file](https://github.com/shakacode/react_on_rails/blob/d08a05776fcbf1aaa0eaff4c4a05c3d3744a4963/docs/basics/i18n.md#L32)):*
> automatically after you finished the following settings.

finished settings is probably not what you're trying to say.

maybe you mean "configured the following settings"

---

*[lib/react_on_rails/locales_to_js.rb, line 21 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-Ka1KcfeULO4kw1sFpFL:-Ka1KcfeULO4kw1sFpFM:beyiyzz) ([raw file](https://github.com/shakacode/react_on_rails/blob/d08a05776fcbf1aaa0eaff4c4a05c3d3744a4963/lib/react_on_rails/locales_to_js.rb#L21)):*
> ```Ruby
>                                           .select { |f| File.exist?(f) }
>       files = files.sort_by { |f| File.mtime(f) }
>       files.last(2).map { |f| File.basename(f) }
> ```

are there a small enough number of files that the performance would never matter?

Here's how we do it for the test helper: https://github.com/shakacode/react_on_rails/blob/master/lib/react_on_rails/test_helper/webpack_assets_status_checker.rb#L28

I actually think that's probably not super efficient either.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642)*

**Comment by justin808** [comment]
Just a few comments remaining.

**Comment by JasonYCHuang** [comment]
Review status: 6 of 11 files reviewed at latest revision, 3 unresolved discussions.

---

*[README.md, line 175 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-Ka1Iv86lD15HMnr-y6W:-KaDQCNs7m2jjKcrTu3A:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/c1c5e351b58c76bb15a0554e57bfd4bdaeb67520/README.md#L175)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

I think we should consider mentioning that we support an optional automatic conversion of Rails *.yml files.
</blockquote></details>

Done.

---

*[docs/basics/i18n.md, line 32 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-Ka1JJFTgRRCnVWxx3AL:-KaDQGTLqXj5wgf-l8tm:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/d08a05776fcbf1aaa0eaff4c4a05c3d3744a4963/docs/basics/i18n.md#L32)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

> automatically after you finished the following settings.

finished settings is probably not what you're trying to say.

maybe you mean "configured the following settings"

</blockquote></details>

Done.

---

*[lib/react_on_rails/locales_to_js.rb, line 21 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-Ka1KcfeULO4kw1sFpFL:-KaDQJM3BZvG2s1Whhov:b-cyhrc1) ([raw file](https://github.com/shakacode/react_on_rails/blob/d08a05776fcbf1aaa0eaff4c4a05c3d3744a4963/lib/react_on_rails/locales_to_js.rb#L21)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

are there a small enough number of files that the performance would never matter?

Here's how we do it for the test helper: https://github.com/shakacode/react_on_rails/blob/master/lib/react_on_rails/test_helper/webpack_assets_status_checker.rb#L28

I actually think that's probably not super efficient either.
</blockquote></details>

Try to improve this, and add test.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642)*

**Comment by justin808** [comment]
Really super awesome!

See comments! so close!

---

Reviewed 5 of 5 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[CHANGELOG.md, line 11 at r6](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-KaGbVOWi4JiimZQyb8v:-KaGbVOWi4JiimZQyb8w:bgwn4b7) ([raw file](https://github.com/shakacode/react_on_rails/blob/24a4d79bc0598d434cdd5c8973ae9ef5ccf2c65a/CHANGELOG.md#L11)):*
> ```Markdown
> - Removed foreman as a dependency. [#678](https://github.com/shakacode/react_on_rails/pull/678) by [x2es](https://github.com/x2es).
> 
> ### Changed
> ```

Added...

---

*[spec/react_on_rails/locales_to_js_spec.rb, line 46 at r6](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-KaGci6KUk---9OdIQmU:-KaGci6KUk---9OdIQmV:b-2klv5s) ([raw file](https://github.com/shakacode/react_on_rails/blob/24a4d79bc0598d434cdd5c8973ae9ef5ccf2c65a/spec/react_on_rails/locales_to_js_spec.rb#L46)):*
> ```Ruby
>       it "doesn't update files" do
>         mt = File.mtime(translations_path)
>         sleep 1
> ```

Let's avoid the sleep!

http://stackoverflow.com/questions/32300446/stubbing-time-now-with-rspec

Set the time to be one minute ago.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-:-KaGcsOpiQumQkxYqpIQ:bfgji2i)*

**Comment by JasonYCHuang** [comment]
@justin808 Thank you for patient reviews.
Hope we can finish the PR this week. :smile: 

---

Review status: 9 of 11 files reviewed at latest revision, 2 unresolved discussions.

---

*[CHANGELOG.md, line 11 at r6](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-KaGbVOWi4JiimZQyb8v:-KaJ8A4xzP188dODMvZ1:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/24a4d79bc0598d434cdd5c8973ae9ef5ccf2c65a/CHANGELOG.md#L11)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Added...
</blockquote></details>

Done.

---

*[spec/react_on_rails/locales_to_js_spec.rb, line 46 at r6](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-KaGci6KUk---9OdIQmU:-KaJ8JoB4iWNO9SpokOv:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/24a4d79bc0598d434cdd5c8973ae9ef5ccf2c65a/spec/react_on_rails/locales_to_js_spec.rb#L46)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Let's avoid the sleep!

http://stackoverflow.com/questions/32300446/stubbing-time-now-with-rspec

Set the time to be one minute ago.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-:-KaJ90abRlDIQsaFstkn:bgoe3dc)*

**Comment by justin808** [comment]
:lgtm:

Awesome!

I'll ship later today.

---

Reviewed 2 of 2 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/642#-:-KaJBugamUmFd1hR6gfM:bi51cmq)*

File: lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt [hunk 1]
```diff
@@ -60,6 +60,12 @@ ReactOnRails.configure do |config|
   config.server_renderer_timeout = 20 # seconds
 
   ################################################################################
+  # I18N OPTIONS
+  ################################################################################
+  # Replace the following line to the location where you keep translation.js & default.js.
+  config.i18n_dir = Rails.root.join("client", "app", "libs", "i18n")
```
Comment (justin808) [inline lib/generators/react_on_rails/templates/base/base/config/initializers/react_on_rails.rb.tt:66]: @JasonYCHuang We added this here, with an invalid directory, and later we abort deploying or testing when this directory does not exist.

**Pull request merged**